src/base/bittorrent/torrentimpl.h

```diff
@@ -300,7 +318,7 @@ namespace BitTorrent
         QString m_name;
         QString m_savePath;
         QString m_category;
-        QSet<QString> m_tags;
+        Tags m_tags;
```

Comment:
I don't mind using `QStringList` instead of `Tags` if we can assume `m_tags` won't be insanely long/many.

Reply:
Wouldn't you benefit from using `std::set`? It is `set` but it is ordered unlike `QSet`.

Reply:
I thought about that but the downside is there will have one more conversion to `QStringList`.

Reply:
>I thought about that but the downside is there will have one more conversion to `QStringList`.

I mean use it everywhere including `Torrent::tags()` etc. Where `QStringList` is really requires? I see there is set-related logic in outer code so you still need to convert tags to set to perform it currently.

Reply:
>Where QStringList is really requires? 

Serialization in WebAPI but I suppose I can provide a `join()` on set to save an operation.

>I mean use it everywhere including Torrent::tags() etc. Where QStringList is really requires? 

Is it acceptable to use std containers in the core classes interface? Is inconsistency a concern?
As for myself, I'm fine with it, mostly.

Reply:
>I suppose I can provide a `join()` on set to save an operation.

👍 
>Is it acceptable to use std containers in the core classes interface?

Recently, I've been thinking that we should not prefer Qt classes (from std and Qt) just out of principle, but rather choose the most rational option on a case-by-case basis, in order to avoid the pointless overhead (of converting back and forth, etc.).

Reply:
In some cases it's good to use custom wrappers to hide implementation behind the convenient interface we like to see in most of qBittorrent code. You don't seem to like that approach, though.

Reply:
>You don't seem to like that approach, though.

Not really, it depends.

Reply:
I'm thinking of adding an `OrderedSet` container which will mimic `QSet` API. This way we can have uniform interface instead of mixing `std` and Qt API styles.

Reply:
You don't even need to provide everything from QSet interface but the subset we really use.